backend/src/routes/api/operator-subscription-status/index.ts

```diff
@@ -0,0 +1,22 @@
+import { KubeFastifyInstance } from '../../../types';
+import { FastifyReply, FastifyRequest } from 'fastify';
+import { secureRoute } from '../../../utils/route-security';
+import { getSubscriptions, isRHOAI } from '../../../utils/resourceUtils';
+
+module.exports = async (fastify: KubeFastifyInstance) => {
+  fastify.get(
+    '/',
+    secureRoute(fastify)(async (request: FastifyRequest, reply: FastifyReply) => {
+      const subscriptions = getSubscriptions();
+      const subNamePrefix = isRHOAI(fastify) ? 'rhods-operator' : 'opendatahub-operator';
+      const operatorSubscriptionStatus = subscriptions.find((sub) =>
+        sub.installedCSV?.includes(subNamePrefix),
+      );
+      if (operatorSubscriptionStatus) {
+        return operatorSubscriptionStatus;
+      }
+      fastify.log.error(`Failed to find operator subscription, ${subNamePrefix}`);
+      reply.send('Not found');
```

Comment:
Shouldn't you return an error instead of a string?

eg. using `http-errors` lib:
```suggestion
      reply.send(reply.send(httpErrors.NotFound););
```